packages/tools/client-debugger/client-debugger-view/src/components/TelemetryView.tsx

```diff
@@ -99,6 +98,147 @@ export function TelemetryView(): React.ReactElement {
+}
+
+/**
+ * A dropdown menu for selecting how many logs to display on the page.
+ */
+function ListLengthSelection(props: ListLengthSelectionProps): React.ReactElement {
+	const { currentLimit, onChangeSelection } = props;
+	const stackTokens: IStackTokens = { childrenGap: 20 };
+
+	// Options formatted for the Fluent Dropdown component
+	const dropdownOptions: { key: number; text: string }[] = [
+		{ key: 50, text: "50" },
+		{ key: 100, text: "100" },
+		{ key: 500, text: "500" },
+		{ key: 1000, text: "1000" },
+	];
+
+	const handleMaxEventChange: DropdownProps["onOptionSelect"] = (event, data) => {
+		onChangeSelection(Number(data.optionText));
+	};
+
+	return (
+		<Stack tokens={stackTokens}>
+			<div style={{ marginLeft: "6px" }}>
+				<h3>Max number of telemetry events to display: </h3>
+				<Dropdown
+					placeholder="Select an option"
+					size="small"
+					style={{ minWidth: "300px", zIndex: "1" }}
+					defaultValue={currentLimit.toString()}
+					// change the number of logs displayed on the page
+					onOptionSelect={handleMaxEventChange}
+				>
+					{dropdownOptions.map((option) => {
+						return (
+							<Option style={{ minWidth: "120px" }} key={option.key}>
+								{option.text}
+							</Option>
+						);
+					})}
+				</Dropdown>
+			</div>
+		</Stack>
+	);
+}
+
+/**
+ * {@link FilteredTelemetryView} input props.
+ */
+interface FilteredTelemetryViewProps {
+	/**
+	 * The current limit (max number of telemetry events to show).
+	 * @defaultValue {@link DEFAULT_PAGE_SIZE}
+	 */
+	telemetryEvents: ITimestampedTelemetryEvent[] | undefined;
+}
+
+function FilteredTelemetryView(props: FilteredTelemetryViewProps): React.ReactElement {
+	const { telemetryEvents } = props;
+	const [eventNameOptions, setEventNameOptions] = useState<string[]>([]);
+	const [selectedCategory, setSelectedCategory] = useState("");
+	const [filteredTelemetryEvents, setFilteredTelemetryEvents] = React.useState<
+		ITimestampedTelemetryEvent[] | undefined
+	>();
+	/**
+	 * Used to store query for the searchable dropdown. The query is used to perform
+	 * partial match searches and will display all events if query is an empty string.
+	 */
+	const [customSearch, setCustomSearch] = React.useState("");
+	const [matchingOptions, setMatchingOptions] = React.useState<string[]>([]);
+
+	React.useEffect(() => {
+		// Create list of all event names
+		setEventNameOptions([
+			...new Set(
+				telemetryEvents?.map((event) =>
+					event.logContent.eventName.slice("fluid:telemetry:".length),
+				),
+			),
+		]);
+		// Initially matching options are all options
+		setMatchingOptions(eventNameOptions.sort());
+		const filteredEvents = getFilteredEvents();
+		setFilteredTelemetryEvents(filteredEvents);
+	}, [telemetryEvents, selectedCategory, customSearch]);
+
+	/**
+	 * Filters all telemetry events based on category and event name
+	 * @returns filtered list of events
+	 */
+	function getFilteredEvents(): ITimestampedTelemetryEvent[] | undefined {
```

Comment:
Nit: Probably best to move this inside of your `useEffect` block. Otherwise we would need to add it as a dependency in the effect dependencies list.